mama/c_cpp/src/c/mama/publisher.h

```diff
@@ -196,8 +283,61 @@ extern mama_status
 mamaPublisher_getTransport (mamaPublisher   publisher,
                             mamaTransport*  transport);
 
+/**
+ * Allocate a publisher events callback structure.
+ * This is done to allow the structure to change more easily.
+ * This structure is disposed by calling mamaPublisherCallbacks_deallocate.
+ *
+ * @param cb Pointer to pointer to a structure.
+ */
+MAMAExpDLL
+extern mama_status
+mamaPublisherCallbacks_allocate (mamaPublisherCallbacks** cb);
+
+/**
+ * Deallocate a publisher events callback structure.
+ * This structure was obtained by calling mamaPublisherCallbacks_allocate.
+ *
+ * @param cb Pointer to a structure.
+ */
+MAMAExpDLL
+extern mama_status
+mamaPublisherCallbacks_deallocate (mamaPublisherCallbacks* cb);
+
+/**
+ * Retrieve the publisher callbacks.
+ *
+ * @param publisher The mamaPublisher from which to send the response.
+ * @param cb Pointer to a cb structure to be populated with the current callbacks.
+ */
+MAMAExpDLL
+extern mama_status
+mamaPublisher_getUserCallbacks (mamaPublisher publisher,
```

Comment:
For getUserCallbacks() the alignment is consistent with the rest of the file, and this is a new method so I can't revert to a previous alignment. Not sure what you are asking for here.